starknet_py/net/full_node_client.py

```diff
@@ -659,6 +657,110 @@ async def get_contract_nonce(
         res = cast(str, res)
         return int(res, 16)
 
+    # ------------------------------- Trace API -------------------------------
+
+    async def trace_transaction(
+        self,
+        tx_hash: Hash,
+    ) -> TransactionTrace:
+        """
+        For a given executed transaction, returns the trace of its execution, including internal calls.
+
+        :param tx_hash: Hash of the executed transaction.
+        :return: Trace of the transaction.
+        """
+        res = await self._client.call(
+            method_name="traceTransaction",
+            params={
+                "transaction_hash": tx_hash,
+            },
+        )
+        return cast(
+            TransactionTrace, TransactionTraceSchema().load(res, unknown=EXCLUDE)
+        )
+
+    async def simulate_transactions(
+        self,
+        transactions: List[AccountTransaction],
+        skip_validate: bool = False,
+        skip_fee_charge: bool = False,
```

Comment:
Are you sure you don't want to pass a Set of `SimulationFlag` instead?
It just seems to me it'd be less obvious to use and understand from documentation, since you don't have a direct connection between `SimulationFlag` Enum which clearly documents all the available flags and the method signature.
Also, if the number of flags used grows, the current approach might quickly get out of hand.

The approach with using set would also make refactoring easier, you would only need to modify a enum without changing function signature and body every time.

Reply:
Yup, you're right ((I stole the idea from starknet.js)), but currently if there are only two flags (and I suspect there won't be more added), the interface is more user-friendly. Passing a bool argument is (at least imo) better than importing an enum just to pass it as an element of an array.  cc @drknzz

Reply:
Agree with the set approach, however we shouldn't expect any more flags here. Current approach is better ui 👍